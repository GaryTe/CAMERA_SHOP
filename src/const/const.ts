export enum Path {
  Catalog = '/',
  Product = 'product/',
  Basket = 'basket',
  Error = '*'
}


export const BACKEND_URL = 'https://camera-shop.accelerator.pages.academy';

export enum APIRoute {
  Cameras = '/cameras',
  Similar = '/similar',
  Reviews = '/reviews',
  Promo = '/promo',
}


export enum NameSpace {
  ListCamera = 'LIST_CAMERA',
  CardProduct = 'CARD_PRODUCT',
  SimilarProduct = 'SIMILAR_PRODUCT',
  Reviews = 'REVIEWS',
  Promo = 'PROMO',
  AddProduct = 'ADD_PRODUCT',
}


export const stringValue = ['price', 'reviewCount', 'descending', 'ascending'];


export enum Category {
  Camera = 'Фотоаппарат',
  Videocamera = 'Видеокамера'
}

export enum TypeCamera {
  Digital = 'Цифровая',
  Membranous = 'Плёночная',
  Instant = 'Моментальная',
  Collection = 'Коллекционная'
}

export enum LevelProduct {
  Zero = 'Нулевой',
  Amateur = 'Любительский',
  Professional = 'Профессиональный'
}


export const HEADLINES: readonly string[] = [
  'Отлично',
  'Хорошо',
  'Нормально',
  'Плохо',
  'Ужасно'
];
