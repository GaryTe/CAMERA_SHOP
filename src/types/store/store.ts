import { store } from '../../store/store/store';
import { listCamera, CardProduct, Reviews, PromoProduct, Review } from '../types-response/types-response';

export type Cameras = {
  cameras: listCamera | null;
  modifiedListProduct: listCamera | null;
}

export type CardCamera = {
  dataCamera: CardProduct | null;
}

export type SimilaCameras = {
  dataSimilaCameras: listCamera | null;
  sortListCamerra: listCamera | null;
  counterMin: number;
  counterMax: number;
}

export type ReviewsCamera = {
  dataReviews: Reviews | Array<Review>;
  resultSortReviews: Reviews | null;
  reviews: Reviews | null;
  counter: number;
}

export type PromoCamera = {
  dataPromoCamera: PromoProduct | null;
}

export type Cart = {
  dataProduct: listCamera | Array<CardProduct>;
  loading: boolean;
}

export type RootState = ReturnType<typeof store.getState>;
