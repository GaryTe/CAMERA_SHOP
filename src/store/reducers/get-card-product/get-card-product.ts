import { createSlice } from '@reduxjs/toolkit';
import { fetchDataCardProduct } from '../../api-actions/api-actions';
import { CardCamera } from '../../../types/store/store';
import { NameSpace } from '../../../const/const';
import { product } from '../../../util/mocks';

const initialState: CardCamera = {
  dataCamera: null
};

export const getCardProduct = createSlice({
  name: NameSpace.CardProduct,
  initialState,
  reducers: {
    functionForTestingReducers: (state) => {
      state.dataCamera = product;
    }
  },
  extraReducers(builder) {
    builder
      .addCase(fetchDataCardProduct.fulfilled, (state, action) => {
        state.dataCamera = action.payload;
      });
  }
});

export const {functionForTestingReducers} = getCardProduct.actions;
