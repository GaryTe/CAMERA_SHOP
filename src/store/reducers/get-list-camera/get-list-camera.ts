import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { fetchDataListCamera } from '../../api-actions/api-actions';
import { down } from '../../../util/util';
import { Cameras } from '../../../types/store/store';
import { NameSpace, stringValue } from '../../../const/const';
import { listCamera } from '../../../types/types-response/types-response';
import { listProduct } from '../../../util/mocks';

const initialState: Cameras = {
  cameras: null,
  modifiedListProduct: null,
};

export const getListCamera = createSlice({
  name: NameSpace.ListCamera,
  initialState,
  reducers: {
    changeDataProduct: (state, action: PayloadAction<listCamera>) => {
      state.modifiedListProduct = action.payload;
    },
    functionForTesting: (state) => {
      state.cameras = listProduct();
      state.modifiedListProduct = listProduct();
    }
  },

  extraReducers(builder) {
    builder
      .addCase(fetchDataListCamera.fulfilled, (state, action) => {
        state.cameras = action.payload;
        state.modifiedListProduct = down(action.payload, stringValue[0]);
      });
  }
});

export const {changeDataProduct, functionForTesting} = getListCamera.actions;
