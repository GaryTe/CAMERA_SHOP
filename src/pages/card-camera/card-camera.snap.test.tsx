import {render} from '@testing-library/react';
import {Provider} from 'react-redux';
import {Router} from 'react-router-dom';
import {createMemoryHistory} from 'history';
import CardCamera from './card-camera';
import { store } from '../../store/store/store';
import { functionForTesting } from '../../store/reducers/get-list-camera/get-list-camera';
import { functionForTestingReducers } from '../../store/reducers/get-card-product/get-card-product';
import { Path } from '../../const/const';

const history = createMemoryHistory();
history.push(`${Path.Catalog}${Path.Product}${1}`);

const children = <CardCamera />;

describe('Component: CardCamera', () => {
  it('should render CardCamera', () => {
    store.dispatch(functionForTesting());
    store.dispatch(functionForTestingReducers());

    const {container} = render(
      <Provider store={store}>
        <Router
          location={history.location}
          navigationType={history.action}
          navigator={history}
        >
          {children}
        </Router>
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
