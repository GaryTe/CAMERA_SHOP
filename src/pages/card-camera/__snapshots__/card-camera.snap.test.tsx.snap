// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CardCamera should render CardCamera 1`] = `
<div>
  <div
    class="error"
  >
    <p
      class="error-message"
    >
      Page does not exist
    </p>
  </div>
</div>
`;
